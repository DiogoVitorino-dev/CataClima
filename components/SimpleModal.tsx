import React from 'react';
import {Modal, StyleSheet, Pressable, useColorScheme} from 'react-native';
import {OpenText} from './StyledText';
import {MaterialCommunityIcons} from '@expo/vector-icons';
import {View} from './Themed';
import Colors from '../constants/Colors';

export default function SimpleModal({
  message,
  modalVisible,
  onCloseModal,
}: {
  message: string;
  modalVisible: boolean;
  onCloseModal: Function;
}) {
  const colorScheme = useColorScheme()
  return (
    <Modal 
    animationType="slide" 
    transparent={true} 
    visible={modalVisible} 
    onRequestClose={() => onCloseModal()} 
    >
      <View style={styles.centeredView} lightColor="transparent" darkColor="transparent">
        <View style={styles.modalView}>
          <OpenText style={styles.modalText}>{message}</OpenText>
          <Pressable style={[styles.button]} onPress={() => onCloseModal()}>
            <MaterialCommunityIcons 
            name="close" 
            size={20} 
            color={Colors[colorScheme ?? 'light'].icon}/>
          </Pressable>
        </View>
      </View>
    </Modal>
  );
}

const styles = StyleSheet.create({
  centeredView: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'flex-end',
    alignItems: 'center',
  },
  modalView: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    margin: 20,
    borderRadius: 20,
    paddingHorizontal: 10,
    alignItems: 'center',
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5,
  },
  button: {
    padding: 10,
    elevation: 2,
  },
  textStyle: {
    fontWeight: 'bold',
    textAlign: 'center',
  },
  modalText: {
    marginHorizontal: 15,
    textAlign: 'center',
  },
});
