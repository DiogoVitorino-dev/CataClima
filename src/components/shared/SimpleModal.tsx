import { FontAwesome } from "@expo/vector-icons";
import React, { useEffect, useState } from "react";
import {
  Modal,
  StyleSheet,
  Pressable,
  ViewStyle,
  StyleProp,
} from "react-native";

import { Text, View } from "./Themed";

import { useAppColor } from "@/hooks/ThemeHooks";

interface IProps {
  message: string;
  visible: boolean;
  onDismiss?: () => void;
  style?: StyleProp<ViewStyle>;
}

export default function SimpleModal({
  message,
  visible,
  onDismiss,
  style,
}: IProps) {
  const color = useAppColor();
  const [visibleState, setVisibleState] = useState(false);

  useEffect(() => setVisibleState(visible), [visible]);

  return (
    <Modal
      animationType="slide"
      transparent
      visible={visibleState}
      onDismiss={onDismiss}
      style={style}
    >

        <View
          style={styles.centeredView}
          lightColor="transparent"
          darkColor="transparent"
        >
          <View
            style={[
              styles.modalView,
              {
                borderColor: color.borderColor,
                borderWidth: StyleSheet.hairlineWidth,
              },
            ]}
          >
            <Text style={styles.modalText}>{message}</Text>
            <Pressable
              accessibilityRole="button"
              focusable
              style={[styles.button]}
              onPress={() => setVisibleState(false)}
            >
              <FontAwesome name="close" size={20} color={color.icon} />
            </Pressable>
          </View>
        </View>
    </Modal>
  );
}

const styles = StyleSheet.create({
  centeredView: {
    flex: 1,
    flexDirection: "column",
    justifyContent: "flex-end",
    alignItems: "center",
  },
  modalView: {
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
    margin: 20,
    borderRadius: 20,
    paddingHorizontal: 10,
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5,
  },
  button: {
    margin: 10,
    elevation: 2,
  },
  textStyle: {
    fontWeight: "bold",
    textAlign: "center",
  },
  modalText: {
    marginHorizontal: 15,
    textAlign: "center",
  },
});
