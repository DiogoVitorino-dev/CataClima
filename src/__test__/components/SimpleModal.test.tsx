import {describe,it, expect, afterEach} from '@jest/globals';
import renderer from 'react-test-renderer'
import SimpleModal from '../../components/SimpleModal';
import { render, fireEvent, cleanup } from '@testing-library/react-native';

describe('SimpleModal component', () => {  
    afterEach(cleanup)

    it('should renders correctly', () => {    
        const tree = renderer.create(<SimpleModal message='test' visible />).toJSON();
        expect(tree).toMatchSnapshot()    
    })
  
    it('should show title', () => {
        const {getByText} = render(<SimpleModal message='test' visible />)
        expect(getByText('test')).toBeTruthy()   
    })

    it('should hide on button click', () => {
        const {getAllByRole,queryAllByRole} = render(<SimpleModal message='test' visible />)

        const button = getAllByRole('button')[1]
        expect(button).toBeDefined()        

        fireEvent.press(button)

        expect(queryAllByRole('button').length).toBeLessThanOrEqual(0)
    })
    
    it('should hide when clicking anywhere on the screen', () => {
        const {getAllByRole,queryAllByRole} = render(<SimpleModal message='test' visible />)

        const button = getAllByRole('button')[0]
        expect(button).toBeDefined()        

        fireEvent.press(button)

        expect(queryAllByRole('button').length).toBeLessThanOrEqual(0)
    })
  
 

})