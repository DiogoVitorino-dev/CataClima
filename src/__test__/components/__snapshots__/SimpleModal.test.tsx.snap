// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleModal component should renders correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  transparent={true}
  visible={false}
/>
`;
